sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/SchemaPruningSuite.scala

```diff
@@ -301,6 +301,24 @@ abstract class SchemaPruningSuite
     checkAnswer(query, Row("Y.", 1) :: Row("X.", 1) :: Row(null, 2) :: Row(null, 2) :: Nil)
   }
 
+  testSchemaPruning("select explode of nested field of array of struct from Generate output") {
```

Comment:
And, why is the test case here? Does it relate to source anything specific?

Reply:
As the current source at `SchemaPruningSuite` doesn't have the data structure suitable for the test, this test doesn't use the source provided here. I put it here because it is highly related to nested column pruning. If we have another proper test suite to put, it is also ok.

Reply:
Sorry, but I agree with @HyukjinKwon 's comment. This will be executed 16 times in `testSchemaPruning` x (ORCv1/ORCv2/Parquetv1/Parquetv2) without any benefit. If you need to add a E2E test, the original place is better than this.

Reply:
Ok. Let me move the test. Thanks.